src/main/java/org/wildfly/security/asn1/ASN1Decoder.java

```diff
@@ -172,21 +172,81 @@
     String decodeBitStringAsString() throws ASN1Exception;
 
     /**
-     * Decode the next ASN.1 element as a printable string.
+     * Decode the next ASN.1 element as a PrintableString.
      *
      * @return the decoded printable string as a string
      * @throws ASN1Exception if the next element is not a printable string
      */
     String decodePrintableString() throws ASN1Exception;
 
     /**
-     * Decode the next ASN.1 element as a printable string.
+     * Decode the next ASN.1 element as a PrintableString.
      *
      * @return the decoded printable string as a byte array
      * @throws ASN1Exception if the next element is not a printable string
      */
     byte[] decodePrintableStringAsBytes() throws ASN1Exception;
 
+    /**
+     * Decode the next ASN.1 element as an UniversalString.
+     *
+     * @return the decoded printable string as a string
+     * @throws ASN1Exception if the next element is not a printable string
+     */
+    default String decodeUniversalString() throws ASN1Exception {
+        throw new UnsupportedOperationException();
```

Comment:
It should be documented that the implementation throws this exception.  Something like this:

```java
@throws UnsupportedOperationException if this implementation does not provide this method
```